ansible/roles/elasticsearch/templates/create_es_snapshot.py.j2

```diff
@@ -18,16 +23,31 @@ def create_snapshot(client):
 
 def delete_old_snapshots(client):
     snaps = client.get('{{ es_repository_name }}', '_all')
-    three_days_ago = (datetime.utcnow() - timedelta(days=3)).isoformat()
-    old_snapshots = [s['snapshot'] for s in snaps['snapshots'] if s['start_time'] < three_days_ago]
+    number_of_days_ago = (datetime.utcnow() - timedelta(days={{elastcsearch_backup_days|default(3)}} )).isoformat()
+    ld_snapshots = [s['snapshot'] for s in snaps['snapshots'] if s['start_time'] < number_of_days_ago]
     for snap in old_snapshots:
         client.delete('{{ es_repository_name }}', snap)
 
 
 def main():
     client = get_snapshot_client()
+# Create Today's Snapshot
     create_snapshot(client)
-    delete_old_snapshots(client)
+# Check if snapshot completed or not
+    itteration = 1
+    while itteration < 5:
+        status = check_snapshot(client)
+        if status == 'SUCESS':
+            # Delete Previous Snapshots
+            delete_old_snapshots(client)
+	    exit(0)
+        else:
+            # Wait
+            time.sleep(300)
+        itteration += 1
+    # Exit after 5 tries.
+    print "Snapshot Not deleted, Since Current Snapshot not completed."
+    exit(1)
```

Comment:
jsut to be clear, so ES doesn't allow us to delete a snapshot as long as there's a snapshot operation running? is that it?

Reply:
Yes. refer: https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-snapshots.html

Reply:
"Only one snapshot process can be executed in the cluster at any time." :(